src/views/DatasetListing/DatasetListingView.js

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+import React, { useMemo, useCallback, useState } from 'react';
+import { Box } from '@mui/material';
+import { ResourceUtils } from '@cosmotech/core';
+import { DatasetListing } from '../../components/DatasetListing';
+import { INGESTION_STATUS } from '../../services/config/ApiConstants';
+import { ACL_PERMISSIONS } from '../../services/config/accessControl';
+import { useDatasetList } from '../DatasetManager/components/DatasetList/DatasetListHook';
+
+export const DatasetListingView = () => {
+  const {
+    datasets,
+    currentDataset,
+    selectDataset,
+    deleteDataset,
+    userPermissionsInCurrentOrganization,
+    isDatasetCopyEnabledInWorkspace,
+  } = useDatasetList();
+
+  const [searchString, setSearchString] = useState('');
+
+  const sortedDatasetList = useMemo(() => ResourceUtils.sortResourceListByName(datasets), [datasets]);
+
+  const filteredDatasets = useMemo(() => {
+    if (!searchString) return sortedDatasetList;
+
+    return sortedDatasetList.filter((dataset) => {
+      const s = searchString.toLowerCase();
+      return dataset.name.toLowerCase().includes(s) || dataset.tags?.some((tag) => tag.toLowerCase().includes(s));
+    });
+  }, [sortedDatasetList, searchString]);
+
+  // CREATE
+  const handleCreateDataset = useCallback(() => {
+    console.log('TODO: open Create Dataset modal');
+  }, []);
+
+  // EDIT
+  const handleEditDataset = useCallback((dataset) => {
+    console.log('TODO: open Edit Dataset modal:', dataset);
+  }, []);
+
+  // COPY
+  const handleCopyDataset = useCallback(
+    (dataset) => {
+      if (!isDatasetCopyEnabledInWorkspace) return;
+      console.log('Copy dataset:', dataset);
+    },
+    [isDatasetCopyEnabledInWorkspace]
+  );
+
+  // SHARE
+  const handleShareDataset = useCallback((dataset) => {
+    console.log('TODO: open Share modal:', dataset);
+  }, []);
+
+  // DELETE
+  const handleDeleteDataset = useCallback(
+    (dataset) => {
+      deleteDataset(dataset.id);
+    },
+    [deleteDataset]
+  );
+
+  // CUSTOM STATUS INFO FOR NEW COMPONENT
+  const mappedDatasets = useMemo(() => {
+    return filteredDatasets.map((dataset) => ({
+      ...dataset,
+      ingestionIcon:
+        dataset.ingestionStatus === INGESTION_STATUS.PENDING
+          ? 'loading'
+          : dataset.ingestionStatus === INGESTION_STATUS.ERROR
+            ? 'error'
+            : 'none',
```

Comment:
`dataset.ingestionStatus` no longer exists in API v5, we must retrieve the id of the runner that generated the dataset, and the status of the runner's last run.
On the `main` branch, there is [a hook](https://github.com/Cosmo-Tech/azure-sample-webapp/blob/main/src/hooks/DatasetRunnerHooks.js#L28-L45) that can help with that